### PR TITLE
Improved handling form level submission attributes (auto-send and auto-delete)

### LINKED ISSUE #5339: Non-empty values passed in auto-send are interpreted as false (forms should not be sent automatically)
#### ODK Collect version
All versions.

#### Problem description
In `auto-send` we support only true/false values. If a user wants to use project settings they have to leave the value empty. That default behavior should be take place also if any unsupported value is passed in `auto-send`. However currently any non-empty value other than true/false is treated as false. Moreover ODK Build adds `default` value by default what is then interpreted by ODK Collect as false.
The same problem might occur in `auto-delete` too.

#### Steps to reproduce the problem
Try to finalize the attached form: 
[AutoSendTest.xlsx](https://github.com/getodk/collect/files/10020535/AutoSendTest.xlsx)

#### Expected behavior
In both `auto-send` and `auto-delete` it should work like:

```
true -> true
false -> false
empty value or any non-empty value -> default behavior (checking project settings)
```

#### Other information 
The problem has been originally reported on the forum: https://forum.getodk.org/t/odk-collect-app-form-auto-send-not-working/39544/11

### PR DESCRIPTION
**Pull request by grzesiek2010**
Closes #5339 

#### What has been done to verify that this works as intended?
I've verified the fix manually and added automated tests.

#### Why is this the best possible solution? Were any other approaches considered?
I was thinking about sanitizing `auto-send` and `auto-delete` before inserting into our database so that values we keep there are only `true`/`false` and then the code that checks it could be a little bit simpler but I didn't have feelings strong enough that it would be better. I would need tests in both places (maybe some would be duplicate) to make sure everything works as expected.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
We need to test auto-send and auto-delete with all possible combinations (project settings + form level settings). Please also review the tests I've added because this pr is a good example of what we talked about recently (reviewing tests in prs)

#### Do we need any specific form for testing your changes? If so, please attach one.
Just forms with `auto-send`/`auto-delete`. See: http://getodk.github.io/xforms-spec/#submission-attributes

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/utilities/InstanceAutoDeleteChecker.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    fun shouldInstanceBeDeleted(
+        formsRepository: FormsRepository,
+        isAutoDeleteEnabledInProjectSettings: Boolean,
+        instance: Instance
+    ): Boolean {
+        formsRepository.getLatestByFormIdAndVersion(instance.formId, instance.formVersion)?.let { form ->
+            return if (isAutoDeleteEnabledInProjectSettings) {
+                form.autoDelete == null || form.autoDelete.trim().lowercase(Locale.US) != "false"
+            } else {
+                form.autoDelete != null && form.autoDelete.trim().lowercase(Locale.US).toBoolean()
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/utilities/InstanceAutoDeleteChecker.kt]: I think an explicit match with `"true"` would be clearer.

**Review by lognaturel — APPROVED** [review]

File: collect_app/src/main/java/org/odk/collect/android/utilities/InstanceAutoDeleteChecker.kt (hunk 1 above)
Comment (grzesiek2010) [reply collect_app/src/main/java/org/odk/collect/android/utilities/InstanceAutoDeleteChecker.kt]: Yeah I agree. Fixed.

**Review by lognaturel — APPROVED** [review]

**Comment by dbemke** [comment]
**Tested with Success!**

Verified on Android 10

**Verified Cases:**
- auto send and auto delete in Collect settings and in form settings
- in settings auto send off/wifi only/ cellular only/ wifi or cellular with/without SIM card with wifi turned on/off
- adding different versions of the same form with auto send and auto delete filled with: default, true/ false with capital/lowercase letters, true/false with the mix of capital and lowercase letters, random words and numbers, an empty space, a space
- sending submissions from different versions of the same form
- hide old versions enabled/disabled
- in settings delete after send enabled/ disabled while in the form auto delete enabled/disabled
- changing settings so that finalized forms get sent automatically after being moved to send finalized form

**Comment by srujner** [comment]
**Tested with Success!**

Verified on Android 13
